deepchecks/vision/datasets/classification/simple.py

```diff
@@ -0,0 +1,205 @@
+    transform : Callable, optional
+        A function/transforms that takes in an image and a label and returns the
+        transformed versions of both.
+        E.g, ``transforms.Rotate``
+    target_transform : Callable, optional
+        A function/transform that takes in the target and transforms it.
+    image_extension : str, default 'JPEG'
+        images format
+    """
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transforms: t.Optional[t.Callable] = None,
+        transform: t.Optional[t.Callable] = None,
+        target_transform: t.Optional[t.Callable] = None,
+        image_extension: str = 'JPEG'
+    ) -> None:
+        self.root_path = Path(root).absolute()
+
+        if not (self.root_path.exists() and self.root_path.is_dir()):
+            raise ValueError(f'{self.root_path} - path does not exist or is not a folder')
+
+        super().__init__(str(self.root_path), transforms, transform, target_transform)
+        self.image_extension = image_extension.lower()
+
+        if train is True:
+            self.images = sorted(self.root_path.glob(f'train/*/*.{self.image_extension}'))
+        else:
+            self.images = sorted(self.root_path.glob(f'test/*/*.{self.image_extension}'))
+
+        if len(self.images) == 0:
+            raise ValueError(f'{self.root_path} - is empty or has incorrect structure')
+
+        # class label -> class index
+        self.classes_map = t.cast(t.Dict[str, int], {
+            img.parent.name: index
+            for index, img in enumerate(self.images)
+        })
+        # class index -> class label
+        self.reverse_classes_map = t.cast(t.Dict[int, str],{
+            v: k
+            for k, v in self.classes_map.items()
+        })
+
+    def __getitem__(self, index: int) -> t.Tuple[pilimage.Image, int]:
+        """Get the image and label at the given index."""
+        image_file = self.images[index]
+        image = cv2.imread(str(image_file))
+        image = pilimage.fromarray(cv2.cvtColor(image, cv2.COLOR_BGR2RGB))
+        return image, self.classes_map[image_file.parent.name]
+
+    def __len__(self) -> int:
+        """Return the number of images in the dataset."""
+        return len(self.images)
+
+
+class SimpleClassificationData(vision.ClassificationData):
+    """Simple ClassificationData type."""
+
+    def batch_to_images(
+        self,
+        batch: t.Tuple[t.Sequence[pilimage.Image], t.Sequence[int]]
+    ) -> t.Sequence[np.ndarray]:
+        """Extract the images from a batch of data."""
+        images, _ = batch
+        return [np.array(img) for img in images]
+
+    def batch_to_labels(
+        self,
+        batch: t.Tuple[t.Sequence[pilimage.Image], t.Sequence[int]]
+    ) -> torch.Tensor:
+        """Extract the labels from a batch of data."""
+        _, labels = batch
+        return torch.Tensor([l for l in labels])
+
+    def infer_on_batch(self, batch, model, device) -> torch.Tensor:
```

Comment:
Should no be implemented (as we really know nothing about the model)

Reply:
removed